components/raftstore/src/router.rs

```diff
@@ -125,6 +129,12 @@ where
     fn release_snapshot_cache(&self);
 }
 
+/// A checker do some checking before routing raft write requests.
+pub trait WritePreChecker: Send + Clone {
```

Comment:
Why not make `RegionInfoAccessor` a field of scheduler directly? `Coprocessor` is supposed to cross abstraction layers.

Reply:
Currently, scheduler only communicates with `Engine`. The engine can be RaftKV and it can also be RocksEngine, BTreeEngine or others. However, RegionInfoAccessor only works with the RaftKV engine. So, I think it would be better to bind RegionInfoAccessor with RaftKV.

Reply:
As I said, `RegionInfoAccessor` is part of coprocessor, it can be used in scheduler.